nuget/lib/dependabot/nuget/file_updater.rb

```diff
@@ -62,9 +66,12 @@ def try_update_projects(dependency)
 
           next unless project_dependencies.any? { |dep| dep.name.casecmp(dependency.name).zero? }
 
-          NativeHelpers.run_nuget_updater_tool(repo_root: repo_contents_path, proj_path: proj_path,
-                                               dependency: dependency, is_transitive: !dependency.top_level?,
-                                               credentials: credentials)
+          call_nuget_updater_tool(dependency, proj_path) unless checked_files.include?(project_file.name)
```

Comment:
This is the primary fix in this PR. If we've already considered a project_file because it's a transitive dependency of one we already ran then we need to skip it. Otherwise we can be running the same projects over and over.